horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/ActionServiceImpl.java

```diff
@@ -127,6 +127,12 @@ public void onNewTest(Test test) {
         executeActions(ActionEvent.TEST_NEW, test.id, test, true);
     }
 
+    @WithRoles(extras = Roles.HORREUM_SYSTEM)
+    @Transactional
+    public void onDatasetLabelsComputed(Integer testId, Object payload) {
+        executeActions(ActionEvent.DATASET_LABELS_COMPUTED, testId, payload, true);
```

Comment:
Here, I am expecting the `payload` to be the `Dataset` obj. So from the `Dataset.EventNew` retrieve the actual dataset object and pass it as payload